### DISCUSSION TITLE
Running a classifier

### DISCUSSION THREAD

**Discussion by garaujo23** [AxisCommunications/acap-computer-vision-sdk-examples#36] [Q&A] [answered]
Hey there, 
I am trying to use the minimal-ml-inference example and use a classifier instead of a detector. I am using the model [mobilenet_v2_1.0_224_quant.tflite](https://github.com/google-coral/edgetpu/blob/master/test_data/mobilenet_v2_1.0_224_quant.tflite) and it appears to work with results labeled MobilenetV2/Predictions/Softmax. If I take the np.argmax(result) it comes out as 918, 766 etc so I assume I need to apply the results with a softmax function? I tried creating softmax using np:
```def softmax(x):
    """Compute softmax values for each sets of scores in x."""
    e_x = np.exp(x - np.max(x))
    return e_x / e_x.sum(axis=0) # only difference```

But i get RuntimeWarning: overflow encountered in exp. Is there an easy way to read the results of the model?

**Comment by Corallo** [comment, marked as answer]
Hello garaujo23!

The model that you linked include indeed a softmax in the end, so **you don't need to apply it after**, you can use the output right away.

Regarding the problem that you have in your softmax implementation (that is not needed in this case), you need to be careful as the output vector is an **unsigned numpy array.** Subtracting np.max(x) will make most of the values go in underflow, this then causes the overflow during the exp operation. You should convert the output array into a signed int before proceeding.

**Reply by garaujo23** [reply]
Thanks @Corallo, I guess the confusion with the softmax came from the values in the array.
![image](https://user-images.githubusercontent.com/25560072/172732606-7c5302dd-d71d-449a-9817-6ab6c2d1e554.png)
When I use argmax and get the index of the highest probability, the value is around 150, 137 etc. Which I don't quite understand because isnt softmax values meant to be between 0-1?

**Reply by Corallo** [reply]
Yes, softmax gives a probability distribution between 0 and 1, but then the model quantize it into integers in the scale 0-255.
In the case of that model, the quantization coefficient is 0.00390625 you can see it by loading your model on  https://netron.app/ and pressing on the last layer.
By multiplying your integer values for that coefficient you get back a float representation between 0 and 1.
